h/static/scripts/tests/controllers/autosuggest-dropdown-controller-test.js

```diff
@@ -0,0 +1,413 @@
+
+          done();
+        });
+    });
+
+    it('can keyboard select', function(done){
+      const onSelectSpy = sinon.spy(defaultConfig, 'onSelect');
+
+      assert.equal(onSelectSpy.callCount, 0);
+
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      syn
+        .click(input)
+        .type('[down][down][enter]', () => {
+
+          assert.equal(onSelectSpy.callCount, 1);
+
+          const selectedItem = onSelectSpy.args[0][0];
+
+          assert.propertyVal(selectedItem, 'title', 'tag:');
+          assert.propertyVal(selectedItem, 'explanation', 'search for annotations with a tag');
+
+          assert.isFalse(isSuggestionContainerVisible(), 'post select hide');
+
+          assert.isFalse(form.onsubmit.called, 'should not submit the form on enter');
+
+          done();
+        });
+    });
+
+    it('can hover items', function(done){
+
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      const list = container.querySelector('.' + defaultConfig.classNames.list);
+
+      syn
+        .click(input, ()=>{
+
+          mouseMove(center(list.childNodes[1]));
+
+          assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 1);
+          assert.isTrue(list.childNodes[1].classList.contains(defaultConfig.classNames.activeItem));
+
+
+          mouseMove(center(list.childNodes[2]));
+
+          assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 1);
+          assert.isTrue(list.childNodes[2].classList.contains(defaultConfig.classNames.activeItem));
+
+          mouseMove(center(input));
+          assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 0);
+
+          done();
+
+        });
+
+    });
+
+    it('correctly sets active elements when swapping between keyboard and mouse setting', function(done){
+
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      const list = container.querySelector('.' + defaultConfig.classNames.list);
+
+      syn
+        .click(input, ()=>{
+
+          mouseMove(center(list.childNodes[2]));
+
+          assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 1);
+          assert.isTrue(list.childNodes[2].classList.contains(defaultConfig.classNames.activeItem));
+
+        })
+        .type('[down]', ()=>{
+          assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 1);
+          assert.isTrue(list.childNodes[3].classList.contains(defaultConfig.classNames.activeItem));
+        })
+        .type('[up][up][up]', ()=>{
+          assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 1);
+          assert.isTrue(list.childNodes[0].classList.contains(defaultConfig.classNames.activeItem));
+
+
+          mouseMove(center(list.childNodes[2]));
+
+          assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 1);
```

Comment:
Doesn't block merge, but I often find it helpful to extract UI queries like this into helper functions, for two reasons:

1. If you change the UI, you only have one place to update in the tests
2. When reading the tests, it is easier to understand something like `assert.equal(countSelectedItems(), 1)` than `assert.lengthOf(list.querySelectorAll('.' + defaultConfig.classNames.activeItem), 1);`